src/app/modules/library/personal-library/personal-library.component.html

```diff
@@ -1,7 +1,46 @@
 <div class="content-block library-teacher__paginate">
   <div fxLayoutAlign="center center" fxLayout.xs="column">
     <a class="mat-caption" href="#" (click)="showInfo($event)" i18n>What are My Units?</a>
+    <select-all-projects-checkbox
+      [selectedAllProjects]="selectedAllProjects()"
+      [selectedSomeProjects]="selectedSomeProjects()"
+      (selectAllProjectsEvent)="selectAllProjects($event)"
+    ></select-all-projects-checkbox>
+    <div *ngIf="numSelectedProjects() > 0" class="secondary-text" i18n>
+      {{ numSelectedProjects() }} selected
+    </div>
+    <button
+      mat-icon-button
+      *ngIf="!showArchived && selectedProjects().length > 0"
+      (click)="archiveSelectedProjects(true)"
+      aria-label="Archive selected units"
+      i18n-aria-label
+      matTooltip="Archive"
+      matTooltipPosition="above"
+      i18n-matTooltip
+    >
+      <mat-icon>archive</mat-icon>
+    </button>
+    <button
+      mat-icon-button
+      *ngIf="showArchived && selectedProjects().length > 0"
+      (click)="archiveSelectedProjects(false)"
+      aria-label="Restore selected units"
+      i18n-aria-label
+      matTooltip="Restore"
+      matTooltipPosition="above"
+      i18n-matTooltip
+    >
+      <mat-icon>unarchive</mat-icon>
+    </button>
```

Comment:
Should we componentize these buttons to ArchiveProjectsButtonComponent? The ArchiveProjectsButtonComponent can show the correct button (archive or unarchive) and also handle the archiving/unarchiving and send the results back to the PersonalLibraryComponent.

Also, we don't need aria-label and i18n-aria-label for these buttons because we already have the matTooltip.

Reply:
Yea we should do this in another issue and then use the component in the unit library and teacher run list.